internal/k8s/k8s.go

```diff
@@ -40,6 +40,8 @@ type Client struct {
 	nodeIndexer  cache.Indexer
 	nodeInformer cache.Controller
 
+	resyncSvcCh <-chan struct{}
```

Comment:
I'll make the commit message more explicit, but it allow to remove k8s.client from SpeakerList, and allow to drop excessive resync

Reply:
Improved the commit message
```
Replace ForceSync() with ResyncSvcCh chan
    
This commit prepare the removal of k8s.Client from SpeakerList
It will make it easier to reasonn about and test
```

Reply:
I just improved the commit message:
```
    Replace ForceSync() with resyncSvcCh chan
    
    This commit prepares the removal of k8s.Client from SpeakerList.
    Before this commit we had a circular dependency between
    the k8s package and the speakerlist package.
    k8s depends on controller, controller depends on speakerlist
    for layer2, and speakerlist depends on k8s for ForceSync().
    To break this loop, instead of calling k8s.ForceSync() from
    speakerlist, we create a resyncSvcCh chan that we pass to
    both k8s and speakerlist.
```

Reply:
https://github.com/metallb/metallb/blob/main/speaker/layer2_controller.go#L96-L115
In layer2 mode, ShouldAnnounce() uses UsableSpeakers() to decide which node should respond to ARP requests, when UsableSpeakers() response changes, we need to re-run ShouldAnnonce() for every services.

MetalLB was initially getting the state only from k8s, since memberlist it has 2 sources, k8s and memberlist. 
Maybe we could hide SpeakerList usage into k8s package instead of into the layer2 controller, but this means some more days of work, and there are other L2 issue that I would like to focus on instead of endlessly working on this one.